scripts/check-update.sh

```diff
@@ -24,10 +24,14 @@ fi
 cp release-$ami_type.auto.pkrvars.hcl release-$ami_type.old.hcl
 ./generate-release-vars.sh $ami_type
 diff_val=$(diff <(grep -v ami_version release-$ami_type.old.hcl) <(grep -v ami_version release-$ami_type.auto.pkrvars.hcl))
+# If no difference in dependencies, check for security update
 if [ -z "$diff_val" ]; then
-    Update=$(./scripts/check-update-security.sh $ami_type)
-    if [ "$Update" != "true" ] && [ "$ami_type" != "al1" ]; then
-        Update=$(./scripts/check-update-security.sh "$ami_type"_arm)
+    # al2023 version already generates a diff in dependency file if it has security updates, so no check necessary if al2023
+    if [ "$ami_type" != "al2023" ]; then
```

Comment:
Can `$ami_type` be `al2023_arm`?

Reply:
yup, it can be. In that case, it won't run the check-update-security script, but only go off the diff_val to see if the configuration file of al2023 which includes all of its dependencies has been changed. We just don't want to do a fallback of checking for a security update since it will never have one unless al2023 version is updated, and that is already in the configuration file so it would be detected by diff_val

Reply:
> al2023 version already generates a diff in dependency file if it has security updates, so no check necessary if al2023

Does this mean that `$diff_val` will not be `0` for AL2023? If so, then isn't this check redundant as we reach this line only if `$diff_val` is `0`?

Reply:
So this is just for the case where the `diff_val` is 0 and we are running with `$ami_type` = AL2023, then we don't want to run the check-update-security script. Whereas if its any of the other AMI types, then we do want to run the security script to see if there are security updates

Reply:
What would be the value of `$diff_val` if `$ami_type` is "AL2023" and there is a security update?

Reply:
it will be 1.

Reply:
Cool thanks!